lib/lte_link_control/lte_lc.c

```diff
@@ -891,41 +891,39 @@ int lte_lc_psm_get(int *tau, int *active_time)
 	char active_time_str[9] = {0};
 	char tau_ext_str[9] = {0};
 	char tau_legacy_str[9] = {0};
+	char response[128] = { 0 };
+	const char ch = ',';
+	char *ret;
 
 	if ((tau == NULL) || (active_time == NULL)) {
 		return -EINVAL;
 	}
 
-	/* We expect match on at least two parameters, <Active-Time> and <Periodic-TAU-ext>.
-	 * Three matches will only happen on modem firmwares >= 1.2.0.
+	/* We need to parse the three last parameters, <Active-Time>, <Periodic-TAU-ext> and
+	 * <Periodic-TAU>. This will only work on modem firmwares >= 1.2.0.
 	 */
-	err = nrf_modem_at_scanf("AT%XMONITOR",
-		"%%XMONITOR: "
-		"%*u,"		/* <reg_status> */
-		"%*[^,],"	/* <full_name> */
-		"%*[^,],"	/* <short_name> */
-		"%*[^,],"	/* <plmn> */
-		"%*[^,],"	/* <tac> */
-		"%*u,"		/* <AcT> */
-		"%*u,"		/* <band> */
-		"%*[^,],"	/* <cell_id> */
-		"%*u,"		/* <phys_cell_id> */
-		"%*u,"		/* <EARFCN> */
-		"%*u,"		/* <rsrp> */
-		"%*u,"		/* <snr> */
-		"%*[^,],"	/* <NW-provided_eDRX_value> */
-		"\"%8[0-1]\","	/* <Active-Time> */
-		"\"%8[0-1]\","	/* <Periodic-TAU-ext> */
-		"\"%8[0-1]\"",	/* <Periodic-TAU>, available only for modem fw >= v1.2.0 */
-		active_time_str, tau_ext_str, tau_legacy_str);
+
+	err = nrf_modem_at_cmd(response, sizeof(response), "AT\%%XMONITOR");
 	if (err < 0) {
 		LOG_ERR("AT command failed, error: %d", err);
 		return -EFAULT;
-	} else if (err < 2) {
-		LOG_ERR("Active time and/or TAU interval is missing");
-		return -EBADMSG;
 	}
 
+	/* The last three fields of AT response looks something like this:
+	 * ,"00011110","00000111","01001001"
+	 * Start by locating the last comma of the string.
+	 */
+	ret = strrchr(response, ch);
+
+	/* tau_legacy_str comes after the comma and a quote mark, thus skip two bytes forward */
+	memcpy(tau_legacy_str, ret + 2, 8);
+
+	/* tau_ext_str comes before tau_legacy_str. With a quote mark, reverse 9 bytes */
+	memcpy(tau_ext_str, ret - 9, 8);
```

Comment:
If you index with minus sign, you need to ensure `ret - 9` is not negative. This disappears if you implement Jan Tore's suggestion, which makes sense to me.